components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -57,21 +57,25 @@
         PreparedStatement prepStmt = null;
         ResultSet rs = null;
         List<IdentityProvider> idps = new ArrayList<IdentityProvider>();
+        Connection tmpdbConnection=null;
+
+
         try {
             if (dbConnection == null) {
-                dbConnection = JDBCPersistenceManager.getInstance().getDBConnection();
+                tmpdbConnection= JDBCPersistenceManager.getInstance().getDBConnection();
```

Comment:
Sonar suggested to introduce a new variable, to avoid been reusing the same variable all over again.